src/process.ts

```diff
@@ -0,0 +1,23 @@
+import { Sequence, Execution, fork } from 'effection';
+import { EventEmitter } from  'events';
+
+export abstract class Process extends EventEmitter {
+  private execution?: Execution;
+
+  start(): Promise<any> {
+    return new Promise((ready) => {
+      if(!this.execution) {
+        this.execution = fork(this.run(ready));
```

Comment:
I am somewhat uneasy about this "invisible" fork here.

Reply:
Worth noting how trio (the library which inspires Effection) handles this, in that you must explicitly pass around a `nursery` object, which can be used to spawn off new tasks. This way, you cannot create any forks secretly. Could be worth considering if a similar design is possible in Effection?

Reply:
I don't think that the fork will actually be invisible since it will fork from whatever fork is currently executing. https://github.com/thefrontside/effection.js/blob/master/src/fork.js#L220

I _think_ that the spec implies that the `Promise` setup code passed to the constructor is synchronous and so whatever is the currently executing fork when start is called will be the parent of the `run` call.